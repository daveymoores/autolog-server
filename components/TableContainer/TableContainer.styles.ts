import css from "styled-jsx/css";

export default css`
  .timesheet__table--scrollable-wrapper {
    overflow-x: scroll;
  }

  .timesheet__table--row {
    display: grid;
    grid-template-rows: repeat(1, var(--cellHeight));
    grid-gap: var(--lineWidth);
  }

  .timesheet__table--row:nth-child(2) {
    margin-top: var(--lineWidth);
  }

  @media (prefers-color-scheme: dark) {
    .timesheet__table--container {
      border: var(--lineWidth) solid var(--lightGreen);
    }

    .timesheet__table--scrollable-wrapper {
      background-color: var(--lightGreen);
    }
  }

  @media (prefers-color-scheme: light) {
    .timesheet__table--container {
      border: var(--lineWidth) solid var(--darkGrey);
    }
    .timesheet__table--scrollable-wrapper {
      background-color: var(--darkGrey);
    }
  }
`;
