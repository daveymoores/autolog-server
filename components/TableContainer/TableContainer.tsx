import React from "react";

import { TimesheetDayLog } from "../../types/Timesheet.types";
import HourInput from "../HourInput/HourInput";
import HourSymbol from "../HourSymbol/HourSymbol";

interface TableRow {
  timesheet: TimesheetDayLog[];
}

const TableContainer: React.FC<TableRow> = ({ timesheet }) => {
  const className =
    (length: 31 | 30 | 29 | 28) =>
    (classes: string): string => {
      switch (length) {
        case 31:
          return `${classes} grid-cols-[repeat(31,_minmax(0,_1fr))]`;
        case 30:
          return `${classes} grid-cols-[repeat(30,_minmax(0,_1fr))]`;
        case 29:
          return `${classes} grid-cols-[repeat(29,_minmax(0,_1fr))]`;
        case 28:
          return `${classes} grid-cols-[repeat(28,_minmax(0,_1fr))]`;
      }
    };

  const gridClass = className(timesheet.length as 31 | 30 | 29 | 28);

  return (
    <div className="lg:w-full overflow-scroll lg:overflow-auto bg-slate-900 py-4 lg:bg-transparent lg:py-0">
      <div className="w-[1200px] container max-w-none lg:w-full lg:max-w-screen-lg xl:max-w-screen-xl 2xl:max-w-screen-2xl">
        <div className={gridClass("align-middle p-1 grid")}>
          {timesheet.map(({ hours }, index: number) => {
            return (
              <>{hours ? <HourInput hours={hours} index={index} /> : <div />}</>
            );
          })}
        </div>
        <div className={gridClass("align-middle p-1 bg-black rounded-md grid")}>
          {timesheet.map(({ weekend }, index: number) => (
            <div
              key={index}
              className={`text-sm lg:text-md text-center font-semibold${
                weekend ? " text-slate-700" : ""
              }`}
            >
              {index + 1}
            </div>
          ))}
        </div>
        <div className={gridClass("align-middle p-1 grid")}>
          {timesheet.map(({ hours }, index: number) => {
            return (
              <>
                {hours ? <HourSymbol hours={hours} index={index} /> : <div />}
              </>
            );
          })}
        </div>
      </div>
    </div>
  );
};

export default TableContainer;
