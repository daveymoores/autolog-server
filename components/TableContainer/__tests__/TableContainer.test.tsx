import React from "react";
import renderer from "react-test-renderer";

import TableContainer from "../TableContainer";

describe("TableContainer", () => {
  it("renders TableContainer unchanged", () => {
    const tree = renderer
      .create(
        <TableContainer
          timesheet={[{ user_edited: false, hours: 0, weekend: false }]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
