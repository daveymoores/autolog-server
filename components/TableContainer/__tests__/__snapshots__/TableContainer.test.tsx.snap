// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableContainer renders TableContainer unchanged 1`] = `
<div
  className="jsx-2616247939 jsx-1936843775 timesheet__table--container"
>
  <div
    className="jsx-2616247939 jsx-1936843775 timesheet__table--scrollable-wrapper"
  >
    <div
      className="jsx-2616247939 jsx-1936843775 timesheet__table--row"
    >
      <div
        className="jsx-2108230735 timesheet__table--cell"
      >
        1
      </div>
    </div>
    <div
      className="jsx-2616247939 jsx-1936843775 timesheet__table--row"
    >
      <div
        className="jsx-2108230735 timesheet__table--cell"
      >
        0
      </div>
    </div>
  </div>
</div>
`;
